build_tools/build_llvm.ps1

```diff
@@ -0,0 +1,91 @@
+# Enable strict mode
+$ErrorActionPreference = 'Stop'
+Set-PSDebug -Trace 2
+
+$this_dir = Split-Path -Path $MyInvocation.MyCommand.Path -Parent
+$repo_root = Resolve-Path -Path "$this_dir/.."
+$llvm_dir = Resolve-Path -Path "$repo_root/third_party/iree/third_party/llvm-project/llvm"
+echo "llvm_dir $llvm_dir"
+$build_dir = "$repo_root/llvm-build"
+$install_dir = "$repo_root/llvm-install"
+
+if (-not (Test-Path $build_dir))
+{
+    New-Item -Path $build_dir -ItemType Directory | Out-Null
+}
+$build_dir = Resolve-Path -Path $build_dir
+$cache_dir = $env:cache_dir
+
+if (-not $cache_dir)
+{
+    $cache_dir = "$repo_root/.build-cache"
+    if (-not (Test-Path $cache_dir))
+    {
+        New-Item -Path $cache_dir -ItemType Directory | Out-Null
+    }
+    $cache_dir = Resolve-Path -Path $cache_dir
+}
+echo "Caching to $cache_dir"
+
+if (-not (Test-Path "$cache_dir/ccache"))
+{
+    New-Item -Path "$cache_dir/ccache" -ItemType Directory | Out-Null
+}
+if (-not (Test-Path "$cache_dir/pip"))
+{
+    New-Item -Path "$cache_dir/pip" -ItemType Directory | Out-Null
+}
+
+$python = (Get-Command python -ErrorAction SilentlyContinue).Source
+echo "Using python: $python"
+
+$env:CC = 'clang-cl.exe'
+$env:CXX = 'clang-cl.exe'
+$env:CCACHE_DIR = "$cache_dir/ccache"
+$env:CCACHE_MAXSIZE = '700M'
+$env:CMAKE_C_COMPILER_LAUNCHER = 'ccache'
+$env:CMAKE_CXX_COMPILER_LAUNCHER = 'ccache'
+$env:CCACHE_SLOPPINESS = 'include_file_ctime,include_file_mtime,time_macros'
+
+& ccache -z
+
+& cmake `
+  -GNinja `
+  -DCMAKE_BUILD_TYPE=Release `
+  -DCMAKE_INSTALL_PREFIX=$install_dir `
+  -DCMAKE_OBJECT_PATH_MAX=4096 `
+  -DCMAKE_EXE_LINKER_FLAGS_INIT=-fuse-ld=lld `
+  -DCMAKE_SHARED_LINKER_FLAGS_INIT=-fuse-ld=lld `
+  -DCMAKE_MODULE_LINKER_FLAGS_INIT=-fuse-ld=lld `
```

Comment:
FYI CMake 3.29+ has access to `-DCMAKE_LINKER_TYPE=LLD`: https://cmake.org/cmake/help/latest/variable/CMAKE_LINKER_TYPE.html

(We plan on bumping the CMake minimum in IREE and replacing `-DIREE_ENABLE_LLD=ON` with that official option)